dataikuapi/dss/ml.py

```diff
@@ -1490,13 +1512,471 @@ def __init__(self, raw_settings, hyperparameter_search_params):
+        Sets the time series resampling categorical imputation parameters
+
+        :param method: Imputation method. Valid values are: FAIL_IF_CONFLICTING, DROP_IF_CONFLICTING, MEAN_MODE
+        :type method: str
+        """
+        resampling_params = self.get_resampling_params()
+        assert method in self.DUPLICATE_TIMESTAMPS_HANDLING_METHODS, "Invalid duplicate timestamp handling method, should be in {}".format(self.DUPLICATE_TIMESTAMPS_HANDLING_METHODS)
+
+    @property
+    def forecast_horizon(self):
+        """
+        :return: Number of time steps to be forecast
+        :rtype: int
+        """
+        return self.mltask_settings["predictionLength"]
+
+    @forecast_horizon.setter
+    def forecast_horizon(self, value):
+        """
+        :param value: Number of time steps to be forecast
+        :type value: int
+        """
+        assert isinstance(value, int)
+        self.mltask_settings["predictionLength"] = value
+
+    @property
+    def evaluation_gap(self):
+        """
+        :return: Number of skipped time steps for evaluation
+        :rtype: int
+        """
+        return self.mltask_settings["evaluationParams"]["gapSize"]
+
+    @evaluation_gap.setter
+    def evaluation_gap(self, value):
+        """
+        :param value: Number of skipped time steps for evaluation
+        :type value: int
+        """
+        assert isinstance(value, int)
+        assert value <= self.forecast_horizon, "Gap must be smaller than the forecast horizon"
+        self.mltask_settings["evaluationParams"]["gapSize"] = value
+        self.mltask_settings["evaluationParams"]["testSize"] = self.forecast_horizon - value
+
+    @property
+    def time_variable(self):
+        """
+        :return: Feature used as time variable
+        :rtype: str
+        """
+        return self.mltask_settings["timeVariable"]
+
+    @time_variable.setter
+    def time_variable(self, value):
+        """
+        :param value: Feature used as time variable
+        :type value: str
+        """
+        assert isinstance(value, str)
+        self.mltask_settings["timeVariable"] = value
+
+    @property
+    def timeseries_identifiers(self):
+        """
+        :return: List of features used as time series identifiers
+        :rtype: list
+        """
+        return self.mltask_settings["timeseriesIdentifiers"]
+
+    @timeseries_identifiers.setter
+    def timeseries_identifiers(self, value):
+        """
+        :param value: List of features used as time series identifiers.
+        :type value: list
+        """
+        assert isinstance(value, list)
+        self.mltask_settings["timeseriesIdentifiers"] = value
+
+    @property
+    def quantiles(self):
+        """
+        :return: List of quantiles to forecast
+        :rtype: list
+        """
+        return self.mltask_settings["quantilesToForecast"]
+
+    @quantiles.setter
+    def quantiles(self, values):
+        """
+        :param values: List of quantiles to forecast
```

Comment:
Should document 
- 0.5 requirement
- 0.0001-0.9999 inclusive range
- 1e-4 precision